### PR TITLE
MOSIP-11739 Code fix after testing

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 1]
```diff
@@ -87,6 +88,9 @@
 
 	@Value("${mosip.regproc.workflow-manager.internal.action.eventbus.port}")
 	private String eventBusPort;
+	
+    @Value("${mosip.regproc.workflow-manager.internal.action.max.allowed.iteration}")
```
Comment (vishwa-vyom) [inline registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: max.allowed.iteration - change to max-allowed-iteration

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 2]
```diff
@@ -481,6 +484,19 @@ private void processPauseAndRequestAdditionalInfo(WorkflowInternalActionDTO work
 			workflowActionService.processWorkflowAction(internalRegistrationStatusDtos,
 					WorkflowActionCode.RESUME_PROCESSING.toString());
 		} else {
+			List<AdditionalInfoRequestDto> additionalInfoRequestDtos = additionalInfoRequestService.
+					getAdditionalInfoRequestByRegIdAndProcess(
+						workflowInternalActionDTO.getRid(), workflowInternalActionDTO.getAdditionalInfoProcess());
+			String iteration=env.getProperty("mosip.regproc.workflow-manager.internal.action.max.allowed.iteration." + workflowInternalActionDTO.getAdditionalInfoProcess());
```
Comment (vishwa-vyom) [inline registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: make this also action-max-allowed.<process_name>

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 3]
```diff
@@ -481,6 +484,19 @@ private void processPauseAndRequestAdditionalInfo(WorkflowInternalActionDTO work
 			workflowActionService.processWorkflowAction(internalRegistrationStatusDtos,
 					WorkflowActionCode.RESUME_PROCESSING.toString());
 		} else {
+			List<AdditionalInfoRequestDto> additionalInfoRequestDtos = additionalInfoRequestService.
+					getAdditionalInfoRequestByRegIdAndProcess(
+						workflowInternalActionDTO.getRid(), workflowInternalActionDTO.getAdditionalInfoProcess());
+			String iteration=env.getProperty("mosip.regproc.workflow-manager.internal.action.max.allowed.iteration." + workflowInternalActionDTO.getAdditionalInfoProcess());
+			int maxAllowedIterationPerForAdditionalInfoProcess;
```
Comment (vishwa-vyom) [inline registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: why there is per and for in the name ?

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 4]
```diff
@@ -426,7 +429,7 @@ private void processRestartParentFlow(WorkflowInternalActionDTO workflowInternal
 		registrationStatusDto
 				.setLatestTransactionTypeCode(RegistrationTransactionTypeCode.INTERNAL_WORKFLOW_ACTION.toString());
 		registrationStatusDto.setSubStatusCode(StatusUtil.WORKFLOW_INTERNAL_ACTION_SUCCESS.getCode());
-		registrationStatusService.updateRegistrationStatus(registrationStatusDto, MODULE_ID, MODULE_NAME);
+		registrationStatusService.updateRegistrationStatusForWorkflowEngine(registrationStatusDto, MODULE_ID, MODULE_NAME);
```
Comment (vishwa-vyom) [inline registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java:432]: These changes are already merged, can you properly rebase and raise PR ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java (hunk 1 above)
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: fixed

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java (hunk 2 above)
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: fixed

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java (hunk 3 above)
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: fixed

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java (hunk 4 above)
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java:432]: these changes are missed in previous fix so rebased and added changes

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**